src/reaction/sorption_base.cc

```diff
@@ -119,14 +119,16 @@ SorptionBase::~SorptionBase(void)
   if (data_ != nullptr) delete data_;
 
   VecScatterDestroy(&(vconc_out_scatter));
-  VecDestroy(vconc_solid);
-
-  for (unsigned int sbi = 0; sbi < substances_.size(); sbi++)
-  {
-    //no mpi vectors
-    xfree(conc_solid[sbi]);
+  if (vconc_solid != NULL) {
```

Comment:
Is it possible to replace vconc_solid + conc_solid by VeSeqDouble ? If yes, just add a TODO to the declaration of vconc_solid.